### ISSUE TITLE
Password corrupted?

### ISSUE THREAD

**Issue by RF-I** [giuspen/cherrytree#1754]
I lost power with a cherrytree file open and when I reboot and try to open it again it does not recognize the password and asks again. I know the password is correct because it opens an older backup. Is there a way to recover this .ctz file?

**Comment by giuspen** [comment]
it is the password protected archive to be likely corrupted and not the password itself. You should look at documents with the same name but trailing tildes in the same folder of your password protected document, those are your automatic backups and removing the trailing tildes you should then be able to open with cherrytree (or just extract with https://www.7-zip.org/)

**Comment by RF-I** [comment]
Thank you so much! That worked. I appreciate your excellent support and product and will make a donation today. I will continue to do so with each solved problem or feature added :)

**Comment by giuspen** [comment]
Thanks @RF-I 😉

**Comment by RF-I** [comment]
Quite welcome. Now for that undo problem and you have achieved perfection ;) 

I trust when you say that it is a tough one, but so many other actions (all of them?) happen without a jolting scroll to the top and a redraw that I don't understand how it can be so much different. Everything else changes right before our eyes so that we don't become disoriented. But you of course know best! If that gets fixed I am a big donor and fan :) What is used more often then undo? (I'm a bad typist)

**Comment by giuspen** [comment]
@RF-I I will experiment again with the undo state machine, basically at the moment an undo step is not applying a small delta but applying the whole node. The solution is to try and apply small deltas but this can become very complicated between text, images, codeboxes, tables, that's why I've been trying to keep it (relatively) simple so far

**Comment by RF-I** [comment]
Much appreciated, I think what you're saying is that you are redrawing the whole node, which gets jarring when it's quite a long page and you are working near the middle or end. Personally I would be satisfied if the small deltas were applied with simple text and formatting, I don't care so much about a jolt when it is codeboxes , tables, images etc. Perhaps that will simplify it? 

I would be happy if it simply stayed where I was working and I could see the result right before my eyes similar to when I do something like alt <up or down arrow> to move a line up or down. Thank you so much f looking into it - I just do not trust undo at all because I can't see and confirm what it did, By the time it scrolls back to where it was I forgot what mistake I was correcting. Sorry to get off topic on this thread and thank you for revisiting it.

**Comment by giuspen** [comment]
Yes I understand, I can try to improve the undos of simplest rich text nodes as a first step
